### PR TITLE
clientv3: close keepalive channel if TTL locally exceeded

### PR DESCRIPTION
**Pull request by heyitsanthony**
/cc @nekto0n

### CODE REVIEW

**Comment by gyuho** [comment]
lgtm. defer to @xiang90

File: clientv3/lease.go [hunk 1]
```diff
@@ -84,26 +84,38 @@ type lessor struct {
...
-		donec:      make(chan struct{}),
-		keepAlives: make(map[LeaseID]*keepAlive),
-		remote:     pb.NewLeaseClient(c.conn),
+		donec:                 make(chan struct{}),
+		keepAlives:            make(map[LeaseID]*keepAlive),
+		remote:                pb.NewLeaseClient(c.conn),
+		firstKeepAliveTimeout: c.cfg.DialTimeout + time.Second,
+	}
+	if l.firstKeepAliveTimeout == time.Second {
+		l.firstKeepAliveTimeout = 5 * time.Second
```
Comment (xiang90) [inline clientv3/lease.go]: define 5 seconds?

File: clientv3/lease.go [hunk 2]
```diff
@@ -327,16 +340,39 @@ func (l *lessor) recvKeepAlive(resp *pb.LeaseKeepAliveResponse) {
...
 		}
 	}
 }
 
+// deadlineLoop reaps any keep alive channels that have not recieved a resposne within
+// the lease TTL
+func (l *lessor) deadlineLoop() {
+	for {
+		select {
+		case <-time.After(5 * time.Second):
```
Comment (xiang90) [inline clientv3/lease.go]: probably 1 second to be safer?

Comment (heyitsanthony) [inline clientv3/lease.go]: ok

File: clientv3/lease.go (hunk 1 above)
Comment (heyitsanthony) [inline clientv3/lease.go]: added `defaultTTL` constant

**Comment by xiang90** [comment]
lgtm

**Pull request merged**